src/LineSegment.cpp

```diff
@@ -19,82 +11,113 @@ double LineSegment::length2() const {
     return (end - start).length2();
 }
 
-double LineSegment::slope() const {
-    return (end.y - start.y)/(end.x - start.x);
-}
-
-bool LineSegment::isVertical() const {
-    return (end.x == start.x) && (end.y != start.y);
-}
-
-Vector2 LineSegment::direction() const {
-    return Vector2(end - start);
+bool LineSegment::isPoint() const {
+    return start == end;
 }
 
-double LineSegment::intercept() const {
-    return start.y - this->slope()*start.x;
+bool LineSegment::doesIntersect(const LineSegment &line) const {
+    return intersects(line).has_value();
 }
 
-std::pair<double, double> LineSegment::coefficients() const {
-    return {slope(), intercept()};
+std::optional<Vector2> LineSegment::intersects(const LineSegment &line) const {
+    auto result = Line::intersect(start, end, line.start, line.end);
```

Comment:
I don't see the problem with code dependencies. They are actually good, because if there is a bug then you immediately notice it because it goes wrong in every function (not just one). Furthermore there is no bug, as mentioned in the documentation intersects return false if the Lines are equal.

Reply:
Take the case

LineSegment(Vector2(1,1), Vector2(3,3)), LineSegment(Vector2(2,2),Vector2(4,4)) and check it...

Reply:
They intersect when I add this test case:
TEST(LineIntersectionTests, Rolf) {
    LineSegment line1(Vector2(1,1), Vector2(3,3));
    LineSegment line2(Vector2(2,2),Vector2(4,4));
    EXPECT_TRUE(line1.intersects(line2));
}